crates/vp/src/native_vp.rs

```diff
@@ -68,6 +68,8 @@ where
     pub vp_wasm_cache: CA,
     /// VP evaluator type
     pub eval: PhantomData<EVAL>,
+    /// WASM instructions gas meter kind
+    pub gas_meter_kind: GasMeterKind,
```

Comment:
Do we need this? It looks like the `new` function is always called with `GasMeterKind::MutGlobal`

Reply:
there is a case where we use the other gas meter kind (host fn), when we call the eval vp host fn